Makefile

```diff
@@ -69,111 +72,86 @@ ensure-venv:
 
 .PHONY: html
 html: ensure-venv versions
-	$(SPHINXBUILD) -b html $(ALLSPHINXOPTS) $(BUILDDIR)/html
-	@echo
-	@echo "Build finished. The HTML pages are in $(BUILDDIR)/html."
+	$(SPHINXBUILD) $(ALLSPHINXOPTS)
 
 .PHONY: dirhtml
-dirhtml: ensure-venv versions
-	$(SPHINXBUILD) -b dirhtml $(ALLSPHINXOPTS) $(BUILDDIR)/dirhtml
-	@echo
-	@echo "Build finished. The HTML pages are in $(BUILDDIR)/dirhtml."
+dirhtml: BUILDER = dirhtml
+dirhtml: html
 
 .PHONY: singlehtml
-singlehtml: ensure-venv
-	$(SPHINXBUILD) -b singlehtml $(ALLSPHINXOPTS) $(BUILDDIR)/singlehtml
-	@echo
-	@echo "Build finished. The HTML page is in $(BUILDDIR)/singlehtml."
+singlehtml: BUILDER = singlehtml
+singlehtml: html
 
 .PHONY: pickle
-pickle: ensure-venv
-	$(SPHINXBUILD) -b pickle $(ALLSPHINXOPTS) $(BUILDDIR)/pickle
+pickle: BUILDER = pickle
+pickle: html
 	@echo
 	@echo "Build finished; now you can process the pickle files."
 
 .PHONY: json
-json: ensure-venv
-	$(SPHINXBUILD) -b json $(ALLSPHINXOPTS) $(BUILDDIR)/json
+json: BUILDER = json
+json: html
 	@echo
 	@echo "Build finished; now you can process the JSON files."
 
 .PHONY: htmlhelp
-htmlhelp: ensure-venv
-	$(SPHINXBUILD) -b htmlhelp $(ALLSPHINXOPTS) $(BUILDDIR)/htmlhelp
+htmlhelp: BUILDER = htmlhelp
+htmlhelp: html
 	@echo
 	@echo "Build finished; now you can run HTML Help Workshop with the" \
-	      ".hhp project file in $(BUILDDIR)/htmlhelp."
+	      ".hhp project file in $(BUILDDIR)/$(BUILDER)."
 
 .PHONY: qthelp
-qthelp: ensure-venv
-	$(SPHINXBUILD) -b qthelp $(ALLSPHINXOPTS) $(BUILDDIR)/qthelp
-	@echo
-	@echo "Build finished; now you can run "qcollectiongenerator" with the" \
-	      ".qhcp project file in $(BUILDDIR)/qthelp, like this:"
-	@echo "# qcollectiongenerator $(BUILDDIR)/qthelp/PythonDevelopersGuide.qhcp"
-	@echo "To view the help file:"
-	@echo "# assistant -collectionFile $(BUILDDIR)/qthelp/PythonDevelopersGuide.qhc"
```

Comment:
Should we keep these `echo`s?

Reply:
No, it duplicates output from Sphinx:

# Before

```
...
build succeeded.

You can now run "qcollectiongenerator" with the .qhcp project file in _build/qthelp, like this:
$ qcollectiongenerator _build/qthelp/Python Developer's Guide.qhcp
To view the help file:
$ assistant -collectionFile _build/qthelp/Python Developer's Guide.qhc

Build finished; now you can run qcollectiongenerator with the .qhcp project file in _build/qthelp, like this:
# qcollectiongenerator _build/qthelp/PythonDevelopersGuide.qhcp
To view the help file:
# assistant -collectionFile _build/qthelp/PythonDevelopersGuide.qhc
```

# After

```
...
build succeeded.

You can now run "qcollectiongenerator" with the .qhcp project file in _build/qthelp, like this:
$ qcollectiongenerator _build/qthelp/PythonDevelopersGuide.qhcp
To view the help file:
$ assistant -collectionFile _build/qthelp/PythonDevelopersGuide.qhc
```